simulator/simple_planning_simulator/include/simple_planning_simulator/vehicle_model/sim_model_actuation_cmd.hpp

```diff
@@ -321,6 +326,75 @@ class SimModelActuationCmd : public SimModelInterface
    * @return variable gear ratio
    */
   double calculateVariableGearRatio(const double vel, const double steer_wheel) const;
+
+private:
+  double calcLateralModel(const double steer, const double steer_des, const double vel) override;
+
+  // adaptive gear ratio conversion model
+  double vgr_coef_a_;
+  double vgr_coef_b_;
+  double vgr_coef_c_;
+};
+
+class SimModelActuationCmdMechanical : public SimModelActuationCmdVGR
+{
+public:
+  /**
+   * @brief constructor (mechanical model)
+   * @param [in] vx_lim velocity limit [m/s]
+   * @param [in] steer_lim steering limit [rad]
+   * @param [in] vx_rate_lim acceleration limit [m/ss]
+   * @param [in] steer_rate_lim steering angular velocity limit [rad/ss]
+   * @param [in] wheelbase vehicle wheelbase length [m]
+   * @param [in] dt delta time information to set input buffer for delay
+   * @param [in] accel_delay time delay for accel command [s]
+   * @param [in] acc_time_constant time constant for 1D model of accel dynamics
+   * @param [in] brake_delay time delay for brake command [s]
+   * @param [in] brake_time_constant time constant for 1D model of brake dynamics
+   * @param [in] steer_delay time delay for steering command [s]
+   * @param [in] steer_time_constant time constant for 1D model of steering dynamics
+   * @param [in] steer_bias steering bias [rad]
+   * @param [in] convert_accel_cmd flag to convert accel command
+   * @param [in] convert_brake_cmd flag to convert brake command
+   * @param [in] convert_steer_cmd flag to convert steer command
+   * @param [in] accel_map_path path to csv file for accel conversion map
+   * @param [in] brake_map_path path to csv file for brake conversion map
+   * @param [in] vgr_coef_a coefficient for variable gear ratio
+   * @param [in] vgr_coef_b coefficient for variable gear ratio
+   * @param [in] vgr_coef_c coefficient for variable gear ratio
+   */
+  SimModelActuationCmdMechanical(
+    double vx_lim, double steer_lim, double vx_rate_lim, double steer_rate_lim, double wheelbase,
+    double dt, double accel_delay, double accel_time_constant, double brake_delay,
+    double brake_time_constant, double steer_delay, double steer_time_constant, double steer_bias,
+    bool convert_accel_cmd, bool convert_brake_cmd, std::string accel_map_path,
+    std::string brake_map_path, double vgr_coef_a, double vgr_coef_b, double vgr_coef_c,
+    MechanicalParams mechanical_params);
+
+  /**
+   * @brief update vehicle states
+   * @param [in] dt delta time [s]
+   */
+  void update(const double & dt) override;
+
+  /**
+   * @brief update vehicle states with controller
+   * @details In updateRungeKutta, calcModel is called four times, but the internal state of PID and
+   * Dynamics should not be updated. Therefore, a method is prepared to update the internal state
+   * only once at the end without using the updateRungeKutta of the interface.
+   */
+  void updateRungeKuttaWithController(const double dt, const Eigen::VectorXd & input);
```

Comment:
I override update() and make own `updateRungeKuttaWithController` not to update internal state in PID and Mechanical Dynamics during rungge kutta.

```
  /**
   * @brief update vehicle states with controller
   * @details In updateRungeKutta, calcModel is called four times, but the internal state of PID and
   * Dynamics should not be updated. Therefore, a method is prepared to update the internal state
   * only once at the end without using the updateRungeKutta of the interface.
   */
```